src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/prisonperson/physicalattributes/PhysicalAttributesSyncService.kt

```diff
@@ -22,24 +26,94 @@ class PhysicalAttributesSyncService(
     }
   }
 
-  suspend fun updatePhysicalAttributes(offenderNo: String) {
-    val telemetry = mutableMapOf("offenderNo" to offenderNo)
+  private fun String?.parseFields(): List<String>? =
+    this?.removeSurrounding("[", "]")
+      ?.split(",")
+      ?.toList()
+      ?.map { it.trim() }
 
+  suspend fun updatePhysicalAttributes(offenderNo: String, fields: List<String>? = null) =
     runCatching {
-      dpsApi.getPhysicalAttributes(offenderNo)!!
-        .let { nomisApi.upsertPhysicalAttributes(offenderNo, it.height?.value, it.weight?.value) }
-        .also {
-          telemetry["bookingId"] = it.bookingId.toString()
-          telemetry["created"] = it.created.toString()
-        }
-    }.onSuccess {
-      telemetryClient.trackEvent("physical-attributes-update-success", telemetry)
+      dpsApi.getPhysicalAttributes(offenderNo)
+        ?.also { updateNomis(offenderNo, it, fields) }
+        ?: throw ProfileDetailsSyncException("No physical attributes found for offenderNo: $offenderNo")
     }.onFailure { e ->
-      telemetry["reason"] = e.message ?: "Unknown error"
-      telemetryClient.trackEvent("physical-attributes-update-failed", telemetry)
+      publishTelemetry(
+        type = "physical-attributes-update-error",
+        offenderNo = offenderNo,
+        fields = fields,
+        reason = (e.message ?: "Unknown error"),
+      )
       throw e
     }
+
+  private suspend fun updateNomis(offenderNo: String, dpsAttributes: PhysicalAttributesDto, fields: List<String>?) {
+    if (fields == null || "HEIGHT" in fields || "WEIGHT" in fields) {
+      physicalAttributesNomisApi.upsertPhysicalAttributes(offenderNo, dpsAttributes.height?.value, dpsAttributes.weight?.value)
+        .also {
+          publishTelemetry(
+            type = "physical-attributes-update-success",
+            offenderNo = offenderNo,
+            fields = fields?.filter { it == "HEIGHT" || it == "WEIGHT" },
+            bookingId = it.bookingId,
+            created = it.created,
+          )
+        }
+    }
+    fields?.filterNot { it == "HEIGHT" || it == "WEIGHT" }?.forEach { field ->
+      profileDetailsNomisApi.upsertProfileDetails(offenderNo, field.toNomisProfileType(), dpsAttributes.valueOf(field))
+        .also {
+          publishTelemetry(
+            type = "profile-details-update-success",
+            offenderNo = offenderNo,
+            fields = listOf(field),
+            bookingId = it.bookingId,
+            created = it.created,
+          )
+        }
+    }
+  }
```

Comment:
This is the main change - we update all fields that the event reportrs as changed. This adds the profile details updates.